File: src/main/java/com/fractalwrench/bpp/FileUtils.java
package com.fractalwrench.bpp;

import java.io.File;

final class FileUtils {

    private FileUtils() {
    }

    static File validateFile(String fileName) throws RuntimeException {
        return validateFile(new File(fileName));
    }

    static File validateFile(File file) throws RuntimeException {
        if (file == null) {
            throw new IllegalArgumentException("Cannot read filename 'null'!");
        }
        if (!file.exists()) {
            throw new RuntimeException(String.format("Unable to find file '%s'", file));
        }
        if (!file.canRead()) {
            throw new RuntimeException(String.format("Unable to read file '%s'", file));
        }
        return file;
    }

}

File: src/main/java/com/fractalwrench/bpp/FileUtils.kt
package com.fractalwrench.bpp

import java.io.File

internal object FileUtils {

    @Throws(RuntimeException::class)
    fun validateFile(fileName: String): File {
        return validateFile(File(fileName))
    }

    @Throws(RuntimeException::class)
    fun validateFile(file: File?): File {
        if (file == null) {
            throw IllegalArgumentException("Cannot read filename 'null'!")
        }
        if (!file.exists()) {
            throw RuntimeException(String.format("Unable to find file '%s'", file))
        }
        if (!file.canRead()) {
            throw RuntimeException(String.format("Unable to read file '%s'", file))
        }
        return file
    }

}
